### PR TITLE
GH-6062: Fix group menu bugs + refactoring

### PR DESCRIPTION
**Pull request by schuppell**

### CODE REVIEW

File: projects/core/src/configurator/commons/facade/configurator-groups.service.ts [hunk 1]
```diff
@@ -144,4 +175,27 @@ export class ConfiguratorGroupsService {
...
+    groups: Configurator.Group[],
+    group: Configurator.Group,
+    parentGroup: Configurator.Group
+  ): Configurator.Group {
+    if (groups.includes(group)) {
+      return parentGroup;
+    }
+
+    //Call function recursive until parent group is returned
+    for (let i = 0; i < groups.length; i++) {
```
Comment (ChristophHi) [inline projects/core/src/configurator/commons/facade/configurator-groups.service.ts]: Can we do this in functional style?

File: projects/core/src/configurator/commons/facade/configurator-commons.service.ts [hunk 2]
```diff
@@ -186,6 +186,7 @@ export class ConfiguratorCommonsService {
       configId: configuration.configId,
       groups: [],
       owner: configuration.owner,
+      productCode: configuration.productCode,
```
[the remark this answers is not in the document]
Comment (ChristophHi) [reply projects/core/src/configurator/commons/facade/configurator-commons.service.ts]: I got it, backend request was with undefined

File: projects/core/src/configurator/commons/store/effects/configurator.effect.ts [hunk 3]
```diff
@@ -352,17 +357,18 @@ export class ConfiguratorEffects {
 
   getGroupWithAttributes(groups: Configurator.Group[]): string {
     const groupWithAttributes: Configurator.Group = groups
-      .filter(currentGroup => currentGroup.attributes)
+      .filter(currentGroup => currentGroup.attributes.length > 0)
```
Comment (ChristophHi) [inline projects/core/src/configurator/commons/store/effects/configurator.effect.ts]: This assumes that currentGroup.attributes is always defined, right? Why is currentGroup.attributes not enough?

File: projects/core/src/configurator/commons/facade/configurator-groups.service.ts (hunk 1 above)
Comment (schuppell) [reply projects/core/src/configurator/commons/facade/configurator-groups.service.ts]: Done

File: projects/core/src/configurator/commons/store/effects/configurator.effect.ts (hunk 3 above)
Comment (schuppell) [reply projects/core/src/configurator/commons/store/effects/configurator.effect.ts]: Attributes is always defined from the normalizer, so checking only if attributes is existing is not sufficient.

**Pull request merged**